autoload/go/path.vim

```diff
@@ -151,6 +151,9 @@ function! go#path#CheckBinPath(binpath)
 
     " if it's in PATH just return it
     if executable(binpath)
+        if v:version == 704 && has('patch235')
+            let binpath = exepath(binpath)
+        endif
```

Comment:
This allows us to call the binary from anywhere even on places we are not changing `$PATH`, like on `go#fmt#Format` where we're checking for `goimports` support for `srcdir` or on `go#def#Jump` where we call `guru`. They both fail if you have `goimports` or `guru` in your `$GOBIN` but not in your `$PATH`.